source/NVDAObjects/window/excel.py

```diff
@@ -670,7 +670,7 @@ def _get_excelApplicationObject(self):
 		self.excelApplicationObject=self.excelWorksheetObject.application
 		return self.excelApplicationObject
 
-	re_definedName=re.compile(ur'^((?P<sheet>\w+)!)?(?P<name>\w+)(\.(?P<minAddress>[a-zA-Z]+[0-9]+)?(\.(?P<maxAddress>[a-zA-Z]+[0-9]+)?(\..*)*)?)?$')
+	re_definedName=re.compile(u'^((?P<sheet>(\'[^\']+\'|[^!]+))!)?(?P<name>\w+)(\.(?P<minAddress>[a-zA-Z]+[0-9]+)?(\.(?P<maxAddress>[a-zA-Z]+[0-9]+)?(\..*)*)?)?$')

```

Comment:
As we discussed could we try and split this up to comment each part of this (or at least comment what we can). I think examples of the edge cases that we are trying to match are quite helpful too. In the absence of automated tests, this at least gives us a suite of manual test cases to use the next time that we find we have to modify this.